person: Automatically add an defaultDailyRate to missing rates

FrederikSchlemmer opened the pull request:
This feature lets the user add a default daily rate to each person. 
According to this default daily rate, the missing rates can be automatically be assigned with this rate.
This avoids that the user needs to add each missing rate manually.

@tinne the budget can only be found according to the budget name assigned in the missingDailyRateBean. Can there be multiple budgets with the identical name?
Also, the database migration should be tested. 

Closes #374

tinne commented:
> @tinne the budget can only be found according to the budget name assigned in the missingDailyRateBean. Can there be multiple budgets with the identical name?

There cannot, but there is an issue about it, cf. #415.

tinne reviewed (changes requested):
The basic functionality seems to be “working sometimes”.

I have two issues that I could not easily sort out:

- Beans used in Wicket pages need to be serializable it appears. Many Bean classes are not. Maybe this feature is used in an error situation only.
- A duplication error may happen. Not quite sure how to reproduce. Hope the stack trace helps:

[Log.txt](https://github.com/adessoAG/budgeteer/files/5151493/Log.txt)

FrederikSchlemmer commented:
@tinne we implemented the changes.
We're now searching for a suitable budget with the id. The name could return multiple results because of the unique constraint for each project.
Also, we added the serializable to the bean.

tinne reviewed (changes requested):
Works better, but I can create the following (at first, used-to-be-)mysterious situation:

By editing the hours of a person, I can create a situation where the Daily Rate is deleted, but the system believes it is ok. I.e., for every hour, the last rate assigned to the hour is still assigned, the Edit Person view contains a non-complete list, and still all budget sums are accurate.

Even more distressingly, the situation remains the same when I restart the server (I run the Spring Boot version). And, of course, the database really does not contain the Daily Rate any more. The situation is only cleaned up once I remove and reimport the work records.

Guess this means the daily rate configuration gets duplicated into the work records, maybe for performance reasons. I am quite sure I dislike the situation.

Maybe at least the unassigned rate budget / time frame combinations list should be computed from daily rate and work record _existance_ data only, ignoring RecordEntity.dailyRate.

Small addition: when the combination addition task has successfully completed and no error was given, please rerun the validation. This should clean up the "Missing daily rate for budget..." warnings.

FrederikSchlemmer commented:
> Small addition: when the combination addition task has successfully completed and no error was given, please rerun the validation. This should clean up the "Missing daily rate for budget..." warnings.

The problem with this addition is, that the notifications are based on the database data. That's why when manually adding daily rates this notification also remains.
This addition would need a fundamental change in the notification creation. I think we should address this in a separate pull request.

tinne commented:
OK, then let's leave it this way with this PR.

FrederikSchlemmer commented:
@tinne I just added the functionalitly when the user adds a record manually the feature won't be displayed anymore. This avoids the potential overwriting of manually records by the automatic feature.
